packages/jest-haste-map/src/index.ts

```diff
@@ -264,10 +260,10 @@ export default class HasteMap extends EventEmitter {
       retainAllFiles: options.retainAllFiles,
       rootDir: options.rootDir,
       roots: Array.from(new Set(options.roots)),
-      skipPackageJson: !!options.skipPackageJson,
-      throwOnModuleCollision: !!options.throwOnModuleCollision,
-      useWatchman: options.useWatchman == null ? true : options.useWatchman,
-      watch: !!options.watch,
+      skipPackageJson: options.skipPackageJson || false,
+      throwOnModuleCollision: options.throwOnModuleCollision || false,
+      useWatchman: options.useWatchman ?? true,
+      watch: options.watch || false,
```

Comment:
Hm.. In a way these are breaking. For instance, if someone is passing `1` or `'true'`. It was strange to have `Boolean()` here. Easily can wait for Jest 29.

Reply:
yeah, let's keep the stuff has different semantics here the same, just to be safe